src/EnergyPlus/HeatingCoils.cc

```diff
@@ -2350,9 +2351,15 @@ namespace HeatingCoils {
                     OutletAirHumRat = FullLoadOutAirHumRat;
                     OutletAirTemp = FullLoadOutAirTemp;
                 } else {
-                    OutletAirEnthalpy = PartLoadRat * FullLoadOutAirEnth + (1.0 - PartLoadRat) * InletAirEnthalpy;
-                    OutletAirHumRat = PartLoadRat * FullLoadOutAirHumRat + (1.0 - PartLoadRat) * InletAirHumRat;
-                    OutletAirTemp = PartLoadRat * FullLoadOutAirTemp + (1.0 - PartLoadRat) * InletAirDryBulbTemp;
+                 if (PartLoadRat > 0) {
+                  OutletAirEnthalpy = PartLoadRat * AirMassFlow / MSHPMassFlowRateAver * (FullLoadOutAirEnth - InletAirEnthalpy) + InletAirEnthalpy;
+                  OutletAirHumRat = PartLoadRat * AirMassFlow / MSHPMassFlowRateAver * (FullLoadOutAirHumRat - InletAirHumRat) + InletAirHumRat;
+                  OutletAirTemp = PartLoadRat * AirMassFlow / MSHPMassFlowRateAver * (FullLoadOutAirTemp - InletAirDryBulbTemp) + InletAirDryBulbTemp;
+                 } else {
+                  OutletAirEnthalpy = PartLoadRat * FullLoadOutAirEnth + (1.0 - PartLoadRat) * InletAirEnthalpy;
+                  OutletAirHumRat = PartLoadRat * FullLoadOutAirHumRat + (1.0 - PartLoadRat) * InletAirHumRat;
+                  OutletAirTemp = PartLoadRat * FullLoadOutAirTemp + (1.0 - PartLoadRat) * InletAirDryBulbTemp;
+                 }
```

Comment:
Why is there an equation for Outlet* in the else? If PLR=0, outlet = inlet. I don't know which is faster (having the IF, ELSE, or not having it), but if PLR=0, doesn't the first set of equations still work?

Reply:
@rraustad You are right. When PLR = 0, the outlet should be equal to the inlet. This is my first fix with quick revision to prove my concept. The existing quadratic relationship between system output and part load ratio is not correct. My fix makes a linear relationship as expected. The first set of equations also work with PLR=0. However, since the first set has a denominator of average mass flow rate, it is possible to have divided by zero. The PLR = 0 set is used for protection:

                  OutletAirEnthalpy = InletAirEnthalpy;
                  OutletAirHumRat =  InletAirHumRat;
                  OutletAirTemp = InletAirDryBulbTemp;